frontend/src/app/reports/reports-chart.service.ts

```diff
@@ -0,0 +1,154 @@
+import {Injectable} from '@angular/core';
+import * as c3 from 'c3';
+import {ChartAPI} from 'c3';
```

Comment:
This kind of looks off to me. You've already imported all of c3, but then you re-import the ChartAPI, when it should already be available.

Reply:
ya, but necessary...